### PR TITLE
Feature/loader added Created Spinner and added check if isLoading

### PR DESCRIPTION
**Pull request by NataliiaRodoman**

### CODE REVIEW

**Review by barantarasNEW — APPROVED** [review]

File: src/components/common/Button/Button.tsx [hunk 1]
```diff
@@ -3,18 +3,25 @@ import type {ButtonProps} from '@mui/material';
 
 import StyledButton from './Styles';
 import {CustomButtonProps} from './types';
+import {Progress} from '../Progress';
 
+// type ButtonPropsWithoutColor = Omit<ButtonProps, 'color'>;
+
+// interface ButtonWithLoadingProps extends Omit<CustomButtonProps, 'children'>, ButtonPropsWithoutColor {
+//     isLoading?: boolean;
+// }
```
Comment (nosovandriy) [inline src/components/common/Button/Button.tsx]: Remove commented lines

**Review by nosovandriy — CHANGES_REQUESTED** [review]

**Review by vanyachyzh — APPROVED** [review]

File: src/components/common/loader/Styles.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {styled} from '@mui/material/styles';
+
+export const BoxLoader = styled('div')(({theme}) => ({
+    width: '100vw',
+    height: '100vh',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    textAlign: 'center',
+    backgroundColor: theme.palette.primary.main,
```
Comment (mykhalenych) [inline src/components/common/loader/Styles.ts]: delete this

File: src/layouts/MainLayout/Styles.ts [hunk 3]
```diff
@@ -23,6 +23,8 @@ export const Main = styled('main')(() => ({
     height: 'calc(100vh - 160px)',
     overflow: 'hidden',
     backgroundSize: 'cover',
+    display: 'flex',
+    position: 'relative',
```
Comment (mykhalenych) [inline src/layouts/MainLayout/Styles.ts]: delete display and position here

File: src/pages/Auth/Activation/Activation.tsx [hunk 4]
```diff
@@ -23,7 +29,7 @@ const Activation = () => {
     return (
         <Grid container spacing={2} px={2} justifyContent="center" alignItems="center">
             <Grid item xs={12}>
-                <Typography variant="h3">{t('welcomeToUseeSvit')}</Typography>
+                <Typography variant="h3">{isLoading ? <Progress /> : t('welcomeToUseeSvit')}</Typography>
```
Comment (mykhalenych) [inline src/pages/Auth/Activation/Activation.tsx]: why it is in Typography?

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/components/common/loader/Progress.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React, {useEffect} from 'react';
```
Comment (mykhalenych) [inline src/components/common/loader/Progress.tsx]:
```suggestion
import React, {useEffect, useState} from 'react';
```

File: src/components/common/loader/Progress.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import {CircularProgress} from '@mui/material';
+
+import {BoxLoader} from './Styles';
+
+export const Progress = () => {
+    const [level, setLevel] = React.useState(0);
```
Comment (mykhalenych) [inline src/components/common/loader/Progress.tsx]:
```suggestion
    const [level, setLevel] = useState(0);
```

**Review by mykhalenych — CHANGES_REQUESTED** [review]

**Review by mykhalenych — APPROVED** [review]

**Pull request merged**